bottlecap/src/logs/processor.rs

```diff
@@ -0,0 +1,339 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use serde::Serialize;
+
+use crate::config;
+use crate::telemetry::events::{TelemetryEvent, TelemetryRecord};
+
+const LOGS_SOURCE: &str = "lambda";
+
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct Lambda {
+    pub arn: String,
+    pub request_id: String,
+}
+
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct LambdaMessage {
+    pub message: String,
+    pub lambda: Lambda,
+    pub timestamp: i64,
+    pub status: String,
+}
+
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct IntakeLog {
+    pub message: LambdaMessage,
+    pub hostname: String,
+    pub service: String,
+    #[serde(rename(serialize = "ddtags"))]
+    pub tags: String,
+    #[serde(rename(serialize = "ddsource"))]
+    pub source: String,
+}
+
+#[derive(Clone)]
+pub struct Processor {
+    function_arn: String,
+    datadog_config: Arc<config::Config>,
+}
+
+impl Processor {
+    pub fn new(function_arn: String, datadog_config: Arc<config::Config>) -> Processor {
+        Processor {
+            function_arn,
+            datadog_config,
+        }
+    }
+
+    pub fn process(&self, event: TelemetryEvent) -> Result<IntakeLog, Box<dyn Error>> {
+        let service = self
+            .datadog_config
+            .service
+            .clone()
+            .unwrap_or("".to_string());
+        let tags = self.datadog_config.tags.clone().unwrap_or("".to_string());
+        let mut log = IntakeLog {
+            hostname: self.function_arn.clone(),
```

Comment:
Future possible optimization, (not for now), instead of copying basically immutable strings all the time we can use a Copy On Write, (Cow), [string](https://doc.rust-lang.org/std/borrow/enum.Cow.html).